pkg/asset/installconfig/platformprovisioncheck.go

```diff
@@ -83,6 +84,19 @@ func (a *PlatformProvisionCheck) Generate(ctx context.Context, dependencies asse
 		if err != nil {
 			return err
 		}
+		isAro := false
+		if ic.Config.Azure.ResourceGroupName != "" {
+			isAro, err = client.CheckIfARO(ctx, ic.Config.Azure.ResourceGroupName)
+			if err != nil {
+				return err
+			}
+		}
+
+		if isAro || ic.Config.Publish != types.InternalPublishingStrategy {
+			if ic.Config.Azure.BaseDomainResourceGroupName == "" {
+				return field.Required(field.NewPath("baseDomainResourceGroupName"), "baseDomainResourceGroupName is the resource group name where the azure dns zone is deployed")
+			}
```

Comment:
This doesn't seem unique to ARO? Can we just remove the isARo flag here?

Reply:
Wrong polarity. Fixed it.